CoreCLR failed to build due to error MSB3073 with MSVC on windows

We tried to build CoreCLR and it failed to build due to error MSB3073 with MSVC on windows. It can be first reproduced on fa8383f revision. It should be a CoreCLR source issue. Could you look at this issue? Thanks a lot.

Repro steps:
1.git clone https://github.com/dotnet/coreclr D:\CoreCLR\src
2.Open a clean x86 prompt and browse to D:\CoreCLR\src
3.build x86 release -- /clp:ShowCommandLine

[log_x86_build.log](https://github.com/dotnet/coreclr/files/3251912/log_x86_build.log)


Error info:
ERROR: An error occured when trying to initialize the tools. 
D:\CoreCLR\src\.dotnet\sdk\3.0.100-preview6-011681\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(148,5): error NETSDK1032: The RuntimeIdentifier platform 'win-x64' and the PlatformTarget 'x86' must be compatible. [D:\CoreCLR\src\Tools\ilasm\ilasm.depproj]
D:\CoreCLR\src\Tools\IL.targets(49,5): error MSB3073: The command "D:\CoreCLR\src\Tools\\ilasm\\ilasm -QUIET -NOLOGO -OPTIMIZE -DLL  -OUTPUT=D:\CoreCLR\src\bin\tests\obj\Windows_NT.x86.Release\Managed\Interop\COM\NETServer\NETServer.DefaultInterfaces\NETServer.DefaultInterfaces.dll  NETServer.DefaultInterfaces.il" exited with code 9009.

--- a/build-test.cmd
+++ b/build-test.cmd
@@ -166,6 +166,9 @@ REM ===
 REM =========================================================================================
 
 call "%__ProjectDir%\init-tools.cmd"
+if NOT [%ERRORLEVEL%]==[0] (
+    exit /b %ERRORLEVEL%
+)
 @if defined _echo @echo on
 
 set "__ToolsDir=%__ProjectDir%\Tools"

--- a/init-tools.cmd
+++ b/init-tools.cmd
@@ -1,6 +1,10 @@
 @if not defined _echo @echo off
 setlocal
 
+:: Clear the 'Platform' environment variable for this session
+:: This avoids overriding the default value from BuildTools projects used for initialization
+set Platform=
+
 set INIT_TOOLS_LOG=%~dp0init-tools.log
 if [%PACKAGES_DIR%]==[] set PACKAGES_DIR=%~dp0.packages
 if [%TOOLRUNTIME_DIR%]==[] set TOOLRUNTIME_DIR=%~dp0Tools

